File: Project-Euler/euler23/euler23.java
import java.util.*;

public class euler23
{
	public static void main(String[] args)
	{
		int sum = 0;
		for(int i : get_list(28123))
		{
			sum += i;
		}
		System.out.println(sum);
	}
	
	private static int sum_divisors(int n)
	{
		int sum = 1;
		int limit = (int)Math.floor(n * 0.5) + 1;
		for(int i = 2; i < limit; i++)
		{
			if(n % i == 0)
			{
				sum += i;
			}
		}
		return sum;
	}
	
	private static boolean is_abundant(int n)
	{
		if(sum_divisors(n) > n)
		{
			return true;
		}
		else 
		{
			return false;
		}
	}
	
	private static ArrayList<Integer> find_abundants(int limit)
	{
		ArrayList<Integer> abundants = new ArrayList<>();
		for(int i = 1; i <= limit; i++)
		{
			if(is_abundant(i))
			{
				abundants.add(i);
			}
		}
		return abundants;
	}
	
	private static ArrayList<Integer> get_list(int limit)
	{
		ArrayList<Integer> abds = find_abundants(limit);
		ArrayList<Integer> list = new ArrayList<>();
		for(int i = 1; i < limit; i++)
		{
			list.add(i);
		}
		for(int x : abds)
		{
			for(int y : abds)
			{
				if(x + y >= limit)
				{
					break;
				}
				list.set(x+y-1, 0);
			}
		}
		return list;
	}
}

File: Project-Euler/euler23/euler23.py
import sys
import math

def sum_divisors(n):
  sum = 1
  limit = int(math.floor(n * 0.5) + 1)
  for i in range(2, limit):
    if n % i == 0:
      sum += i
  return sum

def is_abundant(n):
  if sum_divisors(n) > n:
    return True
  else:
    return False

def find_abundants(limit):
  abundants = []
  for i in range(1,limit+1):
    if is_abundant(i):
      abundants.append(i)
  return abundants

def get_list(limit):
  abds = find_abundants(limit)
  list = [i for i in range(1, limit)]
  for x in abds:
    for y in abds:
      if x + y >= limit:
        break
      list[x+y-1] = 0
  return list

print(sum(get_list(28123)))
